irc/password.go

```diff
@@ -25,10 +27,12 @@ func GenerateEncodedPassword(passwd string) (encoded string, err error) {
 
 func DecodePassword(encoded string) (decoded []byte, err error) {
 	if encoded == "" {
```

Comment:
It probably makes sense to remove this whole `if` statement, since BCrypt passwords will be validated below.